include/wil/result.h

```diff
@@ -252,15 +252,17 @@ namespace wil
                 if (ProcessShutdownInProgress())
                 {
                     // There are no other threads to contend with.
-                    if (--m_refCount == 0)
+                    m_refCount = m_refCount - 1;

```

Comment:
This almost certainly does not need to be `volatile`, but I've opted to keep functionality as-is for now and just address the warnings